src/Microsoft.FeatureManagement/IVariantFeatureManager.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System.Threading.Tasks;
+using System.Threading;
+using Microsoft.FeatureManagement.FeatureFilters;
+
+namespace Microsoft.FeatureManagement
+{
+    /// <summary>
+    /// Used to evaluate the variant for a feature or whether a feature is enabled or disabled.
+    /// </summary>
+    public interface IVariantFeatureManager
+    {
+        /// <summary>
+        /// Checks whether a given feature is enabled.
+        /// </summary>
+        /// <param name="feature">The name of the feature to check.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>True if the feature is enabled, otherwise false.</returns>
+        Task<bool> IsEnabledAsync(string feature, CancellationToken cancellationToken = default);
```

Comment:
I don't see value in the default parameter of cancellation token. I'd suggest we stick with just specifying cancellation token as parameter of the interface.

Reply:
I think I prefer the default. Cancellation token should be available to those who want to use it, but for someone who doesn't care about it (/ is just developing), it's an extra hurdle. For example, I just searched for it because I haven't used them yet. Looks like the developer needs to declare a CancellationTokenSource and have it provide a token?

And from my understanding, we're not respecting the cancellation token yet anyway.

Reply:
No, it looks like this

`IsEnabledAsync(feature, CancellationToken.None)`

Reply:
Default parameters are really good when it's adding an overload to an existing method. Here we don't have that problem. Async methods take cancellation token. The default is a bit of bloat on the interface.

Reply:
Oh sorry, mine was if they want to actually utilize the cancellation token / set it up for future utilization. The point I was trying to make is that I needed to learn what it was/how it's used.

If it's a new expected pattern that all async require a CancellationToken, then we can follow suit. From searching around, it seems like most standard libraries offer async with both the `method()` and `method(CancellationToken token)` options. But perhaps they were building off existing methods, as you said. 

You're also opposed to offering both `IsEnabledAsync(feature)` and `isEnabledAsync(feature, CancellationToken token)` separately, right?

Reply:
> You're also opposed to offering both IsEnabledAsync(feature) and isEnabledAsync(feature, CancellationToken token) separately, right?

Any method returning Task should accept a cancellation token.

Existence of methods returning Task without accepting cancellation token can be considered a bug.